thinc/model.py

```diff
@@ -22,11 +23,18 @@ def empty_init(model: "Model", *args, **kwargs) -> "Model":
     return model
 
 
+class ModelThreadState(threading.local):
+    operators: Dict[str, Callable[["Model", "Model"], "Model"]]
+
+    def __init__(self):
+        self.operators = {}
+
+
 class Model(Generic[InT, OutT]):
     """Class for implementing Thinc models and layers."""
 
     global_id: int = 0
-    _thread_local = create_thread_local({"operators": {}})
+    _thread_local = create_thread_local({"operators": {}}, ModelThreadState)
```

Comment:
Do we have to attach this to the `Model` class? We had it there before but it's always added to the vague this-code-is-haunted vibe I had from the threading stuff.

Reply:
We could do it like I did for the backend file with a `get_model_state` helper function that attaches the model local to the thread object itself. It's a little boilerplate code but I think it makes the whole thing less magic because you get a thread local once, and store a reference to it on the thread it's associated with.